test/unittest/readertest.cpp

```diff
@@ -2324,17 +2324,21 @@ TEST(Reader, ParseNanAndInfinity) {
     double nan = std::numeric_limits<double>::quiet_NaN();
     double inf = std::numeric_limits<double>::infinity();
 
+    TEST_NAN_INF("nan", nan);
     TEST_NAN_INF("NaN", nan);
-    TEST_NAN_INF("-NaN", nan);
+    TEST_NAN_INF("-NaN", nan); // Does testing negative NaN makes any sense? NaN stands for 'Not a Number', thus testing its sign might just be meaningless here.
```

Comment:
I think `-NaN` should be invalid

Reply:
Yep, that was my concern too! Readjusted it as well.